### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by lytvyshko**
-  [DEMO LINK](https://lytvyshko.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

**Review by iwanttosmile4u — APPROVED** [review]

File: src/components/CurrentUser/CurrentUser.tsx [hunk 1]
```diff
@@ -1,12 +1,55 @@
...
-    <h3 className="CurrentUser__name">Ervin Howell</h3>
-    <p className="CurrentUser__email">Shanna@melissa.tv</p>
-    <p className="CurrentUser__phone">010-692-6593 x09125</p>
-  </div>
-);
+export const CurrentUser: React.FC<Props> = ({
+  selectedUserId,
+  onHandlerClear,
+}) => {
+  const [user, setUser] = useState<User>();
```
Comment (Denchepornyuk) [inline src/components/CurrentUser/CurrentUser.tsx:14]: user can be null 
```suggestion
  const [user, setUser] = useState<User | null>();
```

File: src/components/TodoList/TodoList.tsx [hunk 2]
```diff
@@ -1,44 +1,115 @@
...
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
+export const TodoList: React.FC<Props> = ({ todos, selectUser }) => {
+  const [titleSubstring, setTitleSubstring] = useState('');
+  const [filterOption, setFilterOption] = useState('all');
```
Comment (Denchepornyuk) [inline src/components/TodoList/TodoList.tsx:12]: would be great to add a type for filterOption, you can create a enum

**Review by denys-cheporniuk — CHANGES_REQUESTED** [review]

File: src/components/CurrentUser/CurrentUser.tsx [hunk 3]
```diff
@@ -1,12 +1,55 @@
...
+
+  useEffect(() => {
+    getUser(selectedUserId)
+      .then(userFromServer => setUser(userFromServer));
```
Comment (Denchepornyuk) [inline src/components/CurrentUser/CurrentUser.tsx:18]: you have to handle errors, if getUser returns an error you should catch error here

**Review by denys-cheporniuk — APPROVED** [review]
you missed this one
https://github.com/mate-academy/react_dynamic-list-of-todos/pull/874#discussion_r904710447

**Pull request closed without merging**